### ISSUE TITLE
Adding tags to filter through when running

### ISSUE THREAD

**Issue by Shom770** [what-the-python/wtpython#122]
There should be a flag allowed to specify what tags you want for the StackOverflow to filter through, or multiple different custom settings, like when running, something like
``wtpython file.py --python --json``, and instead of setting ``tagged`` to ``python`` in ``settings.py``, you could refer to the multiple tags if specified, so like python;json in this case, as StackOverflow uses ``;`` as a separator for "and" with the tags in the API search.

**Comment by pyxiis** [comment]
I think a syntax like `wtpython file.py --tags python json` would make more sense.

**Comment by VagishVela** [comment]
Would there be a benefit to having a config file with options as well? Or is that over-complication?

**Comment by pyxiis** [comment]
Not really overcomplication if it's left optional.

**Comment by briancohan** [comment]
The CLI usage for wtpython is `wtpython [<wtpython options>] <your script> [<your script options>]` so the tags would have to go before `file.py`

The intent (thus far) is to make it as simple as possible to get to some answers by adding `wt` at the beginning of your command and that's it. I almost feel like manually adding tags or other parameters to the search pushes you to the point where you should just launch your browser. I'm certainly not opposed to considering the option but I'm not sure this fits in with the "simplicity" of how wtpython is intended to work.

As for the config file option, that would not work because whatever you put in there would probably get applied to EVERY search. I highly doubt that you're going to want every thing to have "json" added to each query since you're probably not always working with json.

My goal is to parse out information from the tracebacks to determine which modules and packages were used and automatically add them to the query parameters.

**Comment by Shom770** [comment]
While I understand the move behind simplicity for wtpython, it should also have more customizability with the Stack Overflow search so users can possibly set the parameters for the search, instead of most of the parameters being preset

**Comment by Shom770** [comment]
> While I understand the move behind simplicity for wtpython, it should also have more customizability with the Stack Overflow search so users can possibly set the parameters for the search, instead of most of the parameters being preset

Of course, I believe that some of you have mentioned that in the future WTPython will be less reliant on StackOverflow, so that could be a possible negative to implementing this issue

**Comment by briancohan** [comment]
I tagged this to the New Features milestone since it's something worth considering. But at least for now, I think we need to make our code base a little better. So we can revisit this when we get to that milestone for sure. 

I agree it makes sense given the current state of the application. But for now, I'm personally pushing for  getting our code structured better, having a good CI/CD pipeline, and then polishing the UI/UX around the current features. Once that's good to go, we'll have a good framework for adding new data sources and know the best ways to add improvements like this.
